x-pack/platform/plugins/shared/alerting/server/alerts_client/alerts_client.test.ts

```diff
@@ -2605,145 +2672,85 @@ describe('Alerts Client', () => {
               mw2: [alert3.getUuid()],
             });
 
-          const updateSpy = jest
-            // @ts-ignore
-            .spyOn(AlertsClient.prototype, 'updateAlertMaintenanceWindowIds')
-            // @ts-ignore
-            .mockResolvedValueOnce({});
-
-          const result = await alertsClient.updatePersistedAlertsWithMaintenanceWindowIds();
+          const result = await alertsClient.getAlertsToUpdateWithMaintenanceWindows();
 
           expect(alert1.getMaintenanceWindowIds()).toEqual(['mw3', 'mw1']);
           expect(alert2.getMaintenanceWindowIds()).toEqual(['mw3', 'mw1']);
           expect(alert3.getMaintenanceWindowIds()).toEqual(['mw3', 'mw2']);
 
           expect(result).toEqual({
-            alertIds: [alert1.getUuid(), alert2.getUuid(), alert3.getUuid()],
-            maintenanceWindowIds: ['mw3', 'mw1', 'mw2'],
+            [alert1.getUuid()]: ['mw3', 'mw1'],
+            [alert2.getUuid()]: ['mw3', 'mw1'],
+            [alert3.getUuid()]: ['mw3', 'mw2'],
           });
 
-          expect(updateSpy).toHaveBeenLastCalledWith([
-            alert1.getUuid(),
-            alert2.getUuid(),
-            alert3.getUuid(),
-          ]);
+          expect(loggerSpy).toHaveBeenCalledWith(['mw3', 'mw1', 'mw2']);
         });
+      });
 
-        test('should update alerts based on alert id with MW ids when provided with maintenance windows', async () => {
-          maintenanceWindowsService.getMaintenanceWindows.mockReturnValueOnce({
-            maintenanceWindows: [
-              ...getParamsByUpdateMaintenanceWindowIds.maintenanceWindows,
-              { id: 'mw3' } as unknown as MaintenanceWindow,
-            ],
-            maintenanceWindowsWithoutScopedQueryIds: [],
+      describe('getAlertsToUpdateWithLastScheduledActions()', () => {
```

Comment:
looks like there's some logic to check that meta is correctly populated in this function...should we add a test for that? that alerts are excluded if there's no last scheduled action?